src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -199,7 +199,10 @@ private function findViaExpression(ObjectManager $manager, Request $request, Map
         }
 
         $repository = $manager->getRepository($options->class);
-        $variables = array_merge($request->attributes->all(), ['repository' => $repository]);
+        $variables = array_merge($request->attributes->all(), [
+            'repository' => $repository,
+            'request' => $request,
```

Comment:
This could be a possible BC break if there's a request attribute with the same name that's not Symfony's request object, but I don't think that's likely to happen. Another solution would be:

```php
$variables['request'] ??= $request;
```